transaction/transaction.go

```diff
@@ -892,32 +892,42 @@ func parseTxnForeignApps(foreignApps []uint64) (parsed []types.AppIndex) {
 	return
 }
 
-func SetHeader(
-	tx *types.Transaction,
+func SetApplicationTransactionFields(
```

Comment:
These parameters should be part of the `MakeApplicationCall*` functions.

Reply:
I don't think adding 6 more parameters is a good idea.

Reply:
It's not, you're right. But, as discussed, the solution will be a builder pattern. We don't want to introduce a 3rd way of building transactions. So right now we should stick to the old (and I agree that not-so-great) convention until we introduce the builder pattern method here.

Reply:
done.